pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -693,16 +703,34 @@ func generateProwjob(ciopConfig *api.ReleaseBuildConfiguration,
 	}
 	periodic.ExtraRefs = refs
 
-	if err := getCfg.Defaulter().DefaultPeriodic(periodic); err != nil {
+	if err := r.prowConfigGetter.Defaulter().DefaultPeriodic(periodic); err != nil {
 		return nil, fmt.Errorf("failed to default the ProwJob: %w", err)
 	}
 
-	pj := pjutil.NewProwJob(pjutil.PeriodicSpec(*periodic), labels, annotations, pjutil.RequireScheduling(getCfg.Config().Scheduler.Enabled))
+	pj := pjutil.NewProwJob(pjutil.PeriodicSpec(*periodic), labels, annotations, pjutil.RequireScheduling(r.prowConfigGetter.Config().Scheduler.Enabled))
 	pj.Namespace = prpqrNamespace
 
 	return &pj, nil
 }
 
+func (r *reconciler) clusterForJob(jobName string) (string, error) {
+	if time.Now().Add(time.Minute * -15).After(r.jobClusterCache.lastCleared) {
```

Comment:
Do we need to use a cache? The dispatcher is already caching the job configurations.

Reply:
Yes, Jakub recommended we cache the responses at the client level